python-interface/ODrive.py

```diff
@@ -0,0 +1,282 @@
+            mon = self.active_monitorings[index]
+            if mon.slot != index:
+                mon.slot = index
+                self._ttysend("m %s %s %s" % (mon.datatype, mon.idnumber, mon.slot))
+
+            multiplier = int(base_frequency / self.active_monitorings[index].frequency)
+            try:
+                if new_multipliers_highest_indexes[multiplier] < index:
+                    new_multipliers_highest_indexes[multiplier] = index
+            except:
+                new_multipliers_highest_indexes[multiplier] = index
+
+        self.multipliers_highest_indexes = new_multipliers_highest_indexes
+
+    # these functions are used by the auto generated code
+    def _get(self, datatype, idnumber):
+        self._ttysend("g %s %s" % (datatype, idnumber))
+        value = self._ttyread()
+        return self._convert_get(datatype, value)
+
+    def _set(self, datatype, idnumber, value):
+        self._ttysend("s %s %s %s" % (datatype, idnumber, self._convert_set(datatype, value)))
+
+    def _monitor(self, datatype, idnumber, frequency):
+        self.monitoring_lock.acquire()
+        self.active_monitorings = [ m for m in self.active_monitorings if m.datatype != datatype or m.idnumber != idnumber]
+        if frequency > 0:
+            m = Monitoring(datatype, idnumber, frequency)
+            self.active_monitorings.append(m)
+        self._reorder_monitoring()
+        self.monitoring_lock.release()
+        
+    def _convert_get(self, datatype, value):
+        if datatype == 0:
+            return float(value)
+        if datatype == 1:
+            return int(value)
+        if datatype == 2:
+            if int(value) == 1:
+                return True
+            return False
+
+    def _convert_set(self, datatype, value):
+        if datatype == 0:
+            return float(value)
+        if datatype == 1:
+            return int(value)
+        if datatype == 2:
+            if value == True:
+                return 1
+            return 0
+
+    # raw tty functions, lowest level
+
+    def _ttysend(self, data):
+        if DEBUG:
+            print "sending '%s'" % data
+        if self._serial != None:
+            self._serial.write(data)
+        else:
+            print "Serial dummy: %s" % data
+
+    def _ttyread(self):
+        if self._serial != None:
+            result = self._serial.readline()
+        else:
+            result = 1
+        if DEBUG:
+            print "received '%s'" % result
+        return result
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print "Usage: "
+        print "Odrive.py tty <ip:port>"
+        print ""
+        print "tty     : tty device, required"
+        print "ip:port : optional, defaults to 127.0.0.1:12342"
+        print ""
+        exit(1)
+
+    TTY = sys.argv[1]
+    HOST = "127.0.0.1"
+    PORT = 12342
+
+    try:
+        HOST = sys.argv[2].split(":")[0]
+        PORT = int(sys.argv[2].split(":")[1])
+    except:
```

Comment:
Please don't silence KeyboardInterrupt here, I expect the program to die when I press CTRL-C.
Better to be explicit:

except (IndexError, ValueError):